### DISCUSSION TITLE
Machine Controller method parameter clarification

### DISCUSSION THREAD

**Discussion by jetwatki** [cwapi3d/cwapi3dpython#120] [Q&A] [answered]
Hello all,

I am working on some plugins using the C++ API for CadWork, however I am running into difficulty with understanding the parameters for the methods. I believe the Python API follows the same pattern, so I was hoping to get some help here.

1. What does the paramete aMFBVersion represent in this case?
![image](https://github.com/cwapi3d/cwapi3dpython/assets/111385652/d66fc58b-6653-496e-a9c2-cceb08ce6671)

2. What do each of the parameters for this method represent?
![image](https://github.com/cwapi3d/cwapi3dpython/assets/111385652/2daf4600-b0b2-47b6-ab91-c62d15088f6b)

3. Ditto :)
![image](https://github.com/cwapi3d/cwapi3dpython/assets/111385652/679e91fc-6276-4406-8c9a-22d0119bbc75)

I would really appreciate any help!

**Comment by Brunner246** [comment, marked as answer]
This is at the moment not well documented/implemented. But you find the needed answers in `CwAPI3DTypes.h` 
There are the required enums available.
```cpp
  enum weinmannMfbVersion
  {
    wup_2_0 = 20,
    wup_3_1 = 31,
    wup_3_2 = 32,
    wup_3_3 = 33,
    wup_3_4 = 34
  };
  ```
  
  or in python https://docs.cadwork.com/projects/cwapi3dpython/en/latest/documentation/enums/

**Reply by jetwatki** [reply]
Thank you @Brunner246, I really appreciate it!
